adal/tests/Test.ADAL.NET.UIAutomation/ADALMobileTestHelper.cs

```diff
@@ -77,6 +77,38 @@ public static void AcquireTokenInteractiveHelper(ITestController controller, Use
             CoreMobileTestHelper.PerformSignInFlow(controller, user);
         }
 
+        public static void AcquireTokenWithPromptBehaviorAlwaysHelper(ITestController controller, UserQueryParameters userParams)
+        {
+            var user = PrepareForAuthentication(controller, userParams);
+            SetInputData(controller, CoreUiTestConstants.UiAutomationTestClientId, CoreUiTestConstants.MSGraph);
+            PerformSignInFlowWithPromptBehaviorAlways(controller, user);
+        }
+
+        private static void PerformSignInFlowWithPromptBehaviorAlways(ITestController controller, IUser user)
+        {
+            string passwordInputID = string.Empty;
+            string signInButtonID = string.Empty;
+
+            passwordInputID = CoreUiTestConstants.WebPasswordID;
+            signInButtonID = CoreUiTestConstants.WebSubmitID;
+
+            // Acquire token flow with prompt behavior always
+            controller.Tap(CoreUiTestConstants.AcquireTokenWithPromptBehaviorAlwaysID);
+            CoreMobileTestHelper.AcquireToken(controller, user, passwordInputID, signInButtonID);
+
+            // Execute normal Acquire token flow
+            // The AT flow has promptBehavior.Auto, so the user is only prompted when needed
+            // There should be a token in the cache from the previous call, so the UI will
+            // not be shown again.
+            controller.Tap(CoreUiTestConstants.AcquireTokenID);
+
+            // Execute AT flow w/prompt behavior always
+            // The UI should be shown again.
+            controller.Tap(CoreUiTestConstants.AcquireTokenWithPromptBehaviorAlwaysID);
+            CoreMobileTestHelper.AcquireToken(controller, user, passwordInputID, signInButtonID);
+            CoreMobileTestHelper.VerifyResult(controller);
+        }
```

Comment:
you can move this to the CoreMobileTestHelper (and then you can probably see a few additional things you can refactor)